django/db/models/sql/compiler.py

```diff
@@ -1364,15 +1364,30 @@ def as_sql(self):
         opts = self.query.get_meta()
         insert_statement = self.connection.ops.insert_statement(ignore_conflicts=self.query.ignore_conflicts)
         result = ['%s %s' % (insert_statement, qn(opts.db_table))]
-        fields = self.query.fields or [opts.pk]
-        result.append('(%s)' % ', '.join(qn(f.column) for f in fields))
-
-        if self.query.fields:
+        fields = self.query.fields
+        if not self.connection.features.supports_default_keyword_in_insert:
+            from django.db.models.expressions import DB_DEFAULT
+            db_default_fields = [field for field in fields if field.has_db_default()]
+            omit_fields = set()
+            mixed_fields = []
+            for field in db_default_fields:
+                db_defaults = [isinstance(getattr(obj, field.attname), DB_DEFAULT) for obj in self.query.objs]
+                if all(db_defaults):
+                    omit_fields.add(field)
+                elif any(db_defaults):
+                    mixed_fields.append(field)
+            if mixed_fields:
+                raise Exception(f'Cannot bulk_create {opts.model.__name__} with DEFAULT')
+            fields = [field for field in fields if field not in omit_fields]
```

Comment:
This is rather a hack for the fact that sqlite does not support the `DEFAULT` keyword.

Doing it this way does allow `bulk_create` to work on sqlite when for each field that has a db_default, either all values are explicitly provided or none are. If a value is provided for a field on one object, but not on another, we raise an exception.